common/djangoapps/course_modes/views.py

```diff
@@ -38,6 +38,10 @@ class ChooseModeView(View):
 
     """
 
+    @method_decorator(transaction.non_atomic_requests)
+    def dispatch(self, *args, **kwargs):
+        return super(ChooseModeView, self).dispatch(*args, **kwargs)
+
     @method_decorator(login_required)
```

Comment:
@doctoryes Can we put an atomic on the `get` view to maintain the current behaviour?